### ISSUE TITLE
How can I confirm that the AD9850 is in good condition?

### ISSUE THREAD

**Issue by LilyGO** [F4GOJ/AD9850#3]
First of all, I saw that the serial port information is not displayed in your program and you cannot see the health status. Secondly, I don't have an oscilloscope. I saw that many of the routines in Google use oscilloscopes. Can I check the AD9850 without any accessory equipment?

**Comment by DhanushkaPeiris** [comment]
Yes but you can not completely verify using following method. In this method, only you can take an idea about the function of the AD9850. 

**Method:** Take an small speaker and connect it to the sine wave output of the AD9850 module. Then write a firmware for the AD9850 controller (like a microcontroller or Arduino) to sweep the frequencies from 20 Hz to 20,000 Hz and upload that using the programmer. If the AD9850 module is good you can hear a sound from the speaker. Now, you can know whether the module is working or not little bit. Then, if it is working like this, you have to verify the frequency, for that, change the firmware to generate a single frequency of 1 K Hz. Then, install a spectrum analyzer application into your smartphone, run the application, direct the microphone of the mobile towards the speaker and confirm whether you are having a peak at 1 K Hz on the mobile screen. Then, you can confirm that your AD850 is working well for this case. To check the other frequencies, you can change the firmware. 

Disadvantages: In this method you can only check the frequencies from 20 Hz to 20 K Hz and you can not check the waveform. 

**Alternative solution to observe the waveform cheapest way:** There are plenty of applications to convert your smart phones into an oscilloscope. Following a tutorial about that, you can check the waveform right way. But, this may harmful to your mobile phone. Therefore, use suitable voltage divider before input signals into the mobile. I recommend you to use old unused mobile. I can not guarantee or take any responsibility of your mobile safety in this alternate method.
